File: Data Structures & Algorithms/target-sum/submission-10.py
class Solution:
    def findTargetSumWays(self, nums: List[int], target: int) -> int:
        dp = defaultdict(int) 
        dp[0] = 1 

        for num in nums: 
            next_dp = defaultdict(int) 

            for val, count in dp.items(): 
                next_dp[val + num] += count 
                next_dp[val - num] += count 

            dp = next_dp 

        return dp[target]

File: Data Structures & Algorithms/duplicate-integer/submission-10.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {boolean}
     */
    hasDuplicate(nums) {
        const s = new Set()
        for (const num of nums) {
            if (s.has(num)) {
                return true
            } else {
                s.add(num)
            }
        }

        return false
    }
}
